### PR TITLE
[da-vinci][server] Added flag to enable RocksDB blob file for blocked based format

### PR DESCRIPTION
**Pull request by gaojieliu**
## Summary, imperative, start upper case, don't end with a period
This PR adds options to enable BlobDB feature in RocksDB to reduce write amplification for large value stores.
Currently, these options are only valid for block-based format.

New configs:
rocksdb.blob.files.enabled : default false
rocksdb.min.blob.size.in.bytes : default 4KB
rocksdb.blob.file.size.in.bytes : default 256MB
rocksdb.blob.garbage.collection.age.cutoff : default 0.25
rocksdb.blob.garbage.collection.force.threshold : default 0.8
rocksdb.blob.file.starting.level: default 0

This feature is not perfect, as the main goal is to reduce the write amplification, but space amplification will increase and we can tune the above params to find the right balance.

## How was this PR tested?
CI

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/rocksdb/RocksDBStoragePartition.java [hunk 1]
```diff
@@ -401,6 +401,18 @@ protected Options getStoreOptions(StoragePartitionConfig storagePartitionConfig,
       tableConfig.setCacheIndexAndFilterBlocks(rocksDBServerConfig.isRocksDBSetCacheIndexAndFilterBlocks());
       tableConfig.setFormatVersion(rocksDBServerConfig.getBlockBaseFormatVersion());
       options.setTableFormatConfig(tableConfig);
+
+      /**
+       * Only enable blob files for block-based format.
+       */
+      if (rocksDBServerConfig.isBlobFilesEnabled()) {
+        options.setEnableBlobFiles(true);
+        options.setEnableBlobGarbageCollection(true);
```
Comment (FelixGV) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/rocksdb/RocksDBStoragePartition.java:410]: Is this GC useful for batch-only stores? Should it be enabled only post-EOP instead? If the overhead is null or negligible on a batch-only store then no problem in leaving it always on...

Comment (gaojieliu) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/rocksdb/RocksDBStoragePartition.java:410]: RocksDB SST file writer doesn't respect `blob file` option as all the input data are ordered, so there is no compaction.
That means this option doesn't hurt for batch-only stores.

I am planning to add some metrics and tests to validate this new mode.

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestHybrid.java [hunk 2]
```diff
@@ -1716,6 +1717,8 @@ private static VeniceClusterWrapper setUpCluster(boolean enablePartitionWiseShar
           KafkaConsumerService.ConsumerAssignmentStrategy.PARTITION_WISE_SHARED_CONSUMER_ASSIGNMENT_STRATEGY.name());
     }
     cluster.addVeniceServer(new Properties(), serverProperties);
+    // Enable blob files in one server
+    serverProperties.setProperty(ROCKSDB_BLOB_FILES_ENABLED, "true");
```
Comment (majisourav99) [inline internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestHybrid.java:1721]: May be we could add a JMH benchmark test for blob-file mode.

**Review by majisourav99 — APPROVED** [review]
Thanks, LGTM.

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestHybrid.java (hunk 2 above)
Comment (gaojieliu) [reply internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestHybrid.java:1721]: It is hard to test it properly as it is mainly about write amplification.
I would prefer to validate it in a perf test env.

**Review by majisourav99 — APPROVED** [review]

**Pull request merged**